go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -324,6 +440,51 @@ func newSharedGateway(nodeName string, subnets []*net.IPNet, gwNextHops []net.IP
 	// patch-<logical_port_name_of_localnet_port>-to-br-int
 	patchPort := "patch-" + bridgeName + "_" + nodeName + "-to-br-int"
 
+	// add service route to bridge
+	bridgeLink, err := util.LinkSetUp(bridgeName)
+	if err != nil {
+		return nil, fmt.Errorf("unable to find shared gw bridge interface: %s", bridgeName)
+	}
+	for _, subnet := range config.Kubernetes.ServiceCIDRs {
+		nextHops, err := util.MatchIPFamily(utilnet.IsIPv6CIDR(subnet), gwNextHops)
+		if err != nil {
+			klog.Warningf("Unable to find valid next hop for Service CIDR: %s, available next hops: %v",
+				subnet, gwNextHops)
+		}
```

Comment:
L453: should you continue here? as `nexthops` will be nil and you are accessing it at L456 below.

Reply:
yep I missed it. thanks. As you also mentioned to me before, we probably dont even need this service route. As long as we ensure the legacy route pointing to mp0 is deleted, that should be enough. I think the service subnet will then use the default gateway route which is all we need so that br-ex can take the packet and push it into OVN. Thoughts?